docs/generatedcodeguide.rst

```diff
@@ -0,0 +1,228 @@
+=====================
+Chapel Generated Code
+=====================
+
+This page describes exactly what Chapel code the protocol buffer compiler
+generates for protocol definitions using ``proto3`` syntax. You should read
+the `proto3 language guide`_ before reading this document.
+
+Compiler Invocation
+-------------------
+The protocol buffer compiler produces Chapel output when invoked with the ``--chpl_out``
+command-line flag. The parameter to the ``--chpl_out`` option is the directory where
+you want the compiler to write your Chapel output. The compiler creates a single
+source file for each ``.proto`` file input, having a ``.chpl`` extension.
+
+Only ``proto3`` messages are supported by the Chapel code generator. Ensure
+that each ``.proto`` file begins with a declaration of:
+
+.. code-block:: proto
+
+  syntax = "proto3";
+
+Ouput module name
+-----------------
+The name of the ouput file/module will be same as the ``package`` name. If the
+``package`` name is not specified, the module takes the name of the proto
+file with all non-aplhanumeric characters replaced by an ``underscore``.
+
+For example a file called ``address.proto`` without the package specifier will
+result in an ouput file called ``address.chpl`` with the generated code wrapped
+in a module of the same name.(Full implementation is not shown here.)
+
+.. code-block:: chpl
+
+  module address {
+    record messageName {
+    ...
+    }
+  }
+
+The same proto file with the a package declaration ``package myPackage;`` will
+result in a file called ``myPackage.chpl``.
+
+.. code-block:: chpl
+
+  module myPackage {
+    record messageName {
+    ...
+    }
+  }
+
+Messages
+--------
+Given a simple message declaration:
+
+.. code-block:: proto
+  
+  message Foo {
+    int32 num = 1;
+  }
+  
+The protocol buffer compiler generates a record called ``Foo``, which have message
+field initializers and serialization/parsing methods for wire-type encoding.
+
+ .. code-block:: chpl
+  
+  record Foo {
+
+    // Record fields will be generated corresponding to each proto message field.
+    var num: int(32);
+    
+    // Used to store encoded byte stream of unknown fields encountered while parsing.
```

Comment:
I think it might be effective if we have explanation along with the method declaration. I have replaced single-line comments with multi-line, please check if it looks any better.

Reply:
Hmm, yeah, I think I'd also insert more spacing to break up thoughts in the comments, like the additional line break in my example